src/cmd/validators/transfer.rs

```diff
@@ -96,13 +101,24 @@ impl Create {
                 .map(|o| o.to_vec())
                 .unwrap_or_else(Vec::new),
             fee: 0,
-            stake_amount,
+            stake_amount: 0,
             payment_amount: u64::from(self.amount),
             old_owner_signature: vec![],
             new_owner_signature: vec![],
         };
 
-        txn.fee = txn.txn_fee(&get_txn_fees(&client).await?)?;
+        txn.fee = if let Some(fee) = self.fee {
```

Comment:
I realize now that you were probably trying to follow the stylistic precedent from `txn.fee`. The thing is, this is done for a very specific reason her. The transaction fee depends on the size of the serialize protobuf. 

Therefore, you have to assemble the entire transaction, minus signature and fee before setting the fee.

So in this case, you set the stake amount _after_ setting the fee (although, afaik, the compiler could move those around on you here), so the impact that the stake amount has on the transaction won't be accounted for when calculating the fee.

Reply:
Very interesting! I will update accordingly.